src/tools/compiletest/src/main.rs

```diff
@@ -114,6 +114,7 @@ pub fn parse_config(args: Vec<String>) -> Config {
         )
         .optflag("", "quiet", "print one character per test instead of one line")
         .optopt("", "color", "coloring: auto, always, never", "WHEN")
+        .optflag("", "json", "emit json output instead of plaintext output")
```

Comment:
Probably this needs to be added to src/doc/rustc/src/tests/index.md which tries to document available options... but I'm confused, because that implies there's already a `--format json` option. How does that not work for us?

Reply:
The documentation page you linked lists the options of libtest, while I added an option to compiletest.